alghaibb/resume-builder
# Enhance User Registration and Email Verification Processes

## Description
This pull request introduces several key updates to the authentication and user registration workflow of my resume builder (still haven't figured a proper name for it yet) website. The changes aim to streamline user onboarding, improve security, and ensure a robust verification process.

### Key Changes
**1. Email Verification Logic:**
- Implemented a new email verification flow using the `sendVerificationEmail` function which sends an email with a verification link to new users.
- Added a verification token generation mechanism that creates or replaces a user's verification token.

**2. User Registration Form:**
- Updated the `CreateAccountForm` component to handle user registration, including input validation and asynchronous form submission handling.
- Integrated a new toast notification system for immediate user feedback on registration success or failure.

**3. Password Handling:**
- Enhanced password security measures by implementing `bcrypt` for password hashing.
- Added visibility toggle for password and confirm password fields to improve user experience.

**4. Authentication Configurations:**
Configured `NextAuth` for handling credentials-based authentication with enhanced security and user session management.

### Database Models:
- Updated Prisma models for `User` and `VerificationToken` to support new authentication flows.

### How to Test
1. Navigate to the registration page and attempt to create a new user.
2. Check the database to ensure that the user and verification token are created.
